test_core_undefinedfunctions failing from pickling test

```
$ bin/test pickling
============================= test process starts ==============================
executable:         C:\Users\sylee\AppData\Local\Continuum\anaconda3\envs\myEnvironment\python.exe  (3.6.6-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python
numpy:              1.15.0
random seed:        42911788
hash randomization: on (PYTHONHASHSEED=1681503245)

sympy\utilities\tests\test_pickling.py[41] .........Ff..........ff..f..........f
ff..                                                                      [FAIL]

________________________________________________________________________________
_____ sympy\utilities\tests\test_pickling.py:test_core_undefinedfunctions ______
Traceback (most recent call last):
  File "c:\users\sylee\onedrive\documents\github\sympy\sympy\utilities\tests\test_pickling.py", line 154, in test_core_undefinedfunctions
    check(f, exclude=exclude)
  File "c:\users\sylee\onedrive\documents\github\sympy\sympy\utilities\tests\test_pickling.py", line 77, in check
    c(a, b, d1)
  File "c:\users\sylee\onedrive\documents\github\sympy\sympy\utilities\tests\test_pickling.py", line 76, in c
    assert getattr(b, i) == attr, "%s != %s, protocol: %s" % (getattr(b, i), attr, protocol)
AssertionError: {'__doc__': None, '_extra_kwargs': {}, '__module__': None, '_explicit_class_assumptions': {}, 'default_assumptions': {}, '_prop_handler': {'complex': <function Function._eval_is_complex at 0x00000257D7828730>, 'commutative': <function Function._eval_is_commutative at 0x00000257D7828EA0>, 'positive': <function Expr._eval_is_positive at 0x00000257D7828F28>, 'negative': <function Expr._eval_is_negative at 0x00000257D72DF048>}, '_nargs': None} != {'_extra_kwargs': {}, '__module__': None, '__doc__': None, '_explicit_class_assumptions': {}, 'default_assumptions': {}, '_prop_handler': {'complex': <function Function._eval_is_complex at 0x00000257D5945158>, 'commutative': <function Function._eval_is_commutative at 0x00000257D59450D0>, 'positive': <function Expr._eval_is_positive at 0x00000257D580D268>, 'negative': <function Expr._eval_is_negative at 0x00000257D580D2F0>}, '_nargs': None}, protocol: <module 'cloudpickle' from 'C:\\Users\\sylee\\AppData\\Local\\Continuum\\anaconda3\\envs\\myEnvironment\\lib\\site-packages\\cloudpickle\\__init__.py'>

=== tests finished: 33 passed, 1 failed, 7 expected to fail, in 1.87 seconds ===
DO *NOT* COMMIT!
```

My anaconda environment has `cloudpickle 0.5.3` installed.

My local machine fails on this test with the latest 1.2.1.dev version currently up to date with this repository, and I can give you more necessary system information if you need help tracing down this issue.

--- a/.travis.yml
+++ b/.travis.yml
@@ -23,7 +23,7 @@ matrix:
       env:
         - TEST_ASCII="true"
         # space separated list of optional dependencies(conda packages) to install and test
-        - TEST_OPT_DEPENDENCY="numpy scipy gmpy2 matplotlib>=2.2 theano llvmlite autowrap cython wurlitzer python-symengine=0.3.* tensorflow numexpr ipython antlr-python-runtime>=4.7,<4.8 antlr>=4.7,<4.8"
+        - TEST_OPT_DEPENDENCY="numpy scipy gmpy2 matplotlib>=2.2 theano llvmlite autowrap cython wurlitzer python-symengine=0.3.* tensorflow numexpr ipython antlr-python-runtime>=4.7,<4.8 antlr>=4.7,<4.8 cloudpickle"
         - TEST_SAGE="true"
       addons:
         apt:
@@ -36,7 +36,7 @@ matrix:
     - python: 3.6
       env:
         - TEST_ASCII="true"
-        - TEST_OPT_DEPENDENCY="numpy scipy gmpy2 matplotlib theano llvmlite autowrap cython wurlitzer python-symengine=0.3.* tensorflow numexpr ipython antlr-python-runtime>=4.7,<4.8 antlr>=4.7,<4.8"
+        - TEST_OPT_DEPENDENCY="numpy scipy gmpy2 matplotlib theano llvmlite autowrap cython wurlitzer python-symengine=0.3.* tensorflow numexpr ipython antlr-python-runtime>=4.7,<4.8 antlr>=4.7,<4.8 cloudpickle"
       addons:
         apt:
           packages:

--- a/bin/test_travis.sh
+++ b/bin/test_travis.sh
@@ -129,6 +129,9 @@ test_list = [
     # codegen
     'sympy/codegen/',
     'sympy/utilities/tests/test_codegen',
+
+    # cloudpickle
+    'pickling',
 ]
 
 blacklist = [

--- a/sympy/utilities/tests/test_pickling.py
+++ b/sympy/utilities/tests/test_pickling.py
@@ -148,9 +148,9 @@ def test_core_undefinedfunctions():
     f = Function("f")
     # Full XFAILed test below
     exclude = list(range(5))
-    if sys.version_info < (3,):
-        # https://github.com/cloudpipe/cloudpickle/issues/65
-        exclude.append(cloudpickle)
+    # https://github.com/cloudpipe/cloudpickle/issues/65
+    # https://github.com/cloudpipe/cloudpickle/issues/190
+    exclude.append(cloudpickle)
     check(f, exclude=exclude)
 
 @XFAIL

